securetar/__init__.py

```diff
@@ -189,7 +191,7 @@ def _setup_cipher(self) -> None:
             self.securetar_header = SecureTarHeader.from_bytes(self._file)
             cbc_rand = self.securetar_header.cbc_rand
         else:
-            cbc_rand = os.urandom(IV_SIZE)
+            cbc_rand = self._nonce if self._nonce is not None else os.urandom(IV_SIZE)
```

Comment:
What happens if provided nonce is longer than `IV_SIZE`? Should we maybe validate the length of nonce? :thinking: